src/ua.c

```diff
@@ -833,13 +842,18 @@ int ua_connect(struct ua *ua, struct call **callp,
 	pl.p = (char *)dialbuf->buf;
 	pl.l = dialbuf->end;
 
+	if (ua->custom_hdrs)
+		call_set_custom_hdrs(call, ua->custom_hdrs);
+
 	err = call_connect(call, &pl);
 
 	if (err)
 		mem_deref(call);
 	else if (callp)
 		*callp = call;
 
+	ua_set_custom_hdrs(ua, NULL);
+
```

Comment:
I think we should clear the list of custom headers here, this should be done by the application.